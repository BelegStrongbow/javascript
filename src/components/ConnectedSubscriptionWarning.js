import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";
import { defineMessages, FormattedMessage, injectIntl } from "react-intl";
import { MessageIcon, WarningMessage } from "./MessageBoxes";
import exclamationTriangle from "../icons/exclamation-triangle.svg";
import Link from "./Link";

const messages = defineMessages( {
	explanation: {
		id: "connectedSubscriptionWarning.explanation",
		defaultMessage: "This product is part of a subscription bundle. Cancelling this product will also cancel:",
	},
	contactSupport: {
		id: "connectedSubscriptionWarning.contactSupport",
		defaultMessage: "If you want to cancel just this product, {link}.",
	},
	contactSupportLink: {
		id: "connectedSubscriptionWarning.contactSupportLink",
		defaultMessage: "please contact support",
	},
} );

const MessageContainer = styled.div`
	display:block;
`;


class ConnectedSubscriptionWarning extends React.Component {

	render() {
		let subscriptionProductsListItems = this.props.subscriptions.map( subscription =>
			<li key={ subscription.id }>{ subscription.limit } &times; { subscription.name }</li>
		);
		let supprtLink = <Link to={ "mailto:support@yoast.com" }>
			<FormattedMessage { ...messages.contactSupportLink }/>
		</Link>;

		return (
			<WarningMessage>
				<MessageIcon iconSource={ exclamationTriangle }/>
				<MessageContainer>
					<FormattedMessage { ...messages.explanation }/>
					<ul>
						{ subscriptionProductsListItems }
					</ul>
					<FormattedMessage { ...messages.contactSupport } values={ { link: supprtLink } }/>
				</MessageContainer>
			</WarningMessage>
		);
	}
}

ConnectedSubscriptionWarning.propTypes = {
	subscriptions: PropTypes.array,
};

ConnectedSubscriptionWarning.defaultProps = {
	subscriptions: [],
};

export default injectIntl( ConnectedSubscriptionWarning );
