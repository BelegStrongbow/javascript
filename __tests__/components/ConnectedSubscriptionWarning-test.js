import React from 'react';
import { createComponentWithIntl } from "../../utils";
import ConnectedSubscriptionWarning from "../../src/components/ConnectedSubscriptionWarning";


let subscriptions = [
	{
		id: "product 1",
		limit: 1,
		name:"some test product 1"
	},
	{
		id: "product 2",
		limit: 1,
		name:"some test product 2"
	},
	{
		id: "product 3",
		limit: 2,
		name:"some test product 3"
	},
	{
		id: "product 4",
		limit: 5,
		name:"some test product 4"
	}
];


test( 'the sites component matches the snapshot', () => {
	const component = createComponentWithIntl(
		<ConnectedSubscriptionWarning subscriptions={ subscriptions }/>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
