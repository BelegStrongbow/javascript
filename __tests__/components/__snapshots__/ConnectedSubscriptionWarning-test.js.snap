// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ConnectedSubscriptionWarning component matches the snapshot 1`] = `
.c1 {
  padding: 1em;
  background-color: #ffeb3b;
  margin: 0.5em 0 0 0;
  overflow: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 {
  background-color: #ffeb3b;
}

.c2 {
  background-position: 50%;
  background-repeat: no-repeat;
  background-size: 2em;
  background-image: url( exclamation-triangle.svg );
  height: 2em;
  min-width: 2em;
  margin: 0 1em;
}

.c3 {
  display: block;
}

@media screen and ( max-width:800 ) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    text-align: left;
  }
}

<div
  className="c0 c1"
>
  <span
    className="c2"
  />
  <div
    className="c3"
  >
    <span>
      This product is part of a subscription bundle. Cancelling this product will also cancel:
    </span>
    <ul>
      <li>
        1
         × 
        some test product 1
      </li>
      <li>
        1
         × 
        some test product 2
      </li>
      <li>
        2
         × 
        some test product 3
      </li>
      <li>
        5
         × 
        some test product 4
      </li>
    </ul>
    <span>
      If you want to cancel just this product, 
      <a
        href="mailto:support@yoast.com"
      >
        <span>
          please contact support
        </span>
      </a>
      .
    </span>
  </div>
</div>
`;
